### DISCUSSION TITLE
Passing the source input name to a transform

### DISCUSSION THREAD

**Discussion by victorjkhalaf** [vectordotdev/vector#16486] [Q&A] [answered]
Is it possible to pass the source name in a transform?
For example I have multiple sources for my application (nginx, php error logs, application logs), each have their own source definition. In one transform that takes in all of these as inputs, I want to append some Datadog tags to help identify and process them in Datadog Logs. 
For `ddsource` I need to pass the source input's name. In other words, the value should be either webapp, access or php.
```
[transforms.ddtags]
type = "remap"
inputs = ["webapp", "access", "php"]
source = '''
.ddsource = ???
```
I just don't know if it's possible to pass the current event's source to the .ddsource tag.

I apologize in advance if this was asked before or it's an easy docs search, I couldn't find anything from my own searching.

**Comment by neuronull** [comment, marked as answer]
Hello,

We don't store the component name in the event metadata.

One way to achieve what you are after could be to have a separate transform for each source, which sets the `.ddsource` based on the single input component.

**Reply by jszwedko** [reply]
Related: https://github.com/vectordotdev/vector/issues/11544

**Reply by victorjkhalaf** [reply]
Ah darn! was hoping not to resort to duplication, but if that's what has to be done then so be it. 
Thanks for the input
